src/liballoc/collections/btree/node.rs

```diff
@@ -585,9 +611,10 @@ impl<'a, K, V> NodeRef<marker::Mut<'a>, K, V, marker::Internal> {
         unsafe {
             slice_insert(self.keys_mut(), 0, key);
             slice_insert(self.vals_mut(), 0, val);
+            let internal_node = &mut *self.as_internal_mut();
```

Comment:
Elsewhere you have a comment for why it is safe to create a reference to the entire node. Could you also add one here?